src/services/stringCompletions.ts

```diff
@@ -484,7 +485,17 @@ function getStringLiteralCompletionEntries(sourceFile: SourceFile, node: StringL
             const existing = new Set(namedImportsOrExports.elements.map(n => moduleExportNameTextEscaped(n.propertyName || n.name)));
             const uniques = exports.filter(e => e.escapedName !== InternalSymbolName.Default && !existing.has(e.escapedName));
             return { kind: StringLiteralCompletionKind.Properties, symbols: uniques, hasIndexSignature: false };
-
+        case SyntaxKind.BinaryExpression:
+            if ((parent as BinaryExpression).operatorToken.kind === SyntaxKind.InKeyword) {
+                const type = typeChecker.getTypeAtLocation((parent as BinaryExpression).right);
+                const properties = type.isUnion() ? typeChecker.getAllPossiblePropertiesOfTypes(type.types) : type.getApparentProperties();
+                return {
+                    kind: StringLiteralCompletionKind.Properties,
+                    symbols: properties.filter(prop => !prop.valueDeclaration || !isPrivateIdentifierClassElementDeclaration(prop.valueDeclaration)),
+                    hasIndexSignature: false,

```

Comment:
Does this mean that we will aggressively complete in this position? Does this need to return true if any type has a string/number index signature?

Reply:
This is only used to return value for `isNewIdentifierLocation` and I don't quite understand what this is meant to represent. IIRC it, at times, works inconsistently anyway. I'd love to be able to tell what should be returned for it here but I simply just don't know ;p